genesyscloud/routing_settings/resource_genesyscloud_routing_settings.go

```diff
@@ -0,0 +1,216 @@
+	"fmt"
+	"log"
+	"terraform-provider-genesyscloud/genesyscloud/provider"
+	"terraform-provider-genesyscloud/genesyscloud/util"
+	"terraform-provider-genesyscloud/genesyscloud/util/constants"
+	"terraform-provider-genesyscloud/genesyscloud/util/resourcedata"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+
+	"terraform-provider-genesyscloud/genesyscloud/consistency_checker"
+
+	resourceExporter "terraform-provider-genesyscloud/genesyscloud/resource_exporter"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v130/platformclientv2"
+)
+
+func getAllRoutingSettings(_ context.Context, clientConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+	resources := make(resourceExporter.ResourceIDMetaMap)
+	resources["0"] = &resourceExporter.ResourceMeta{Name: "routing_settings"}
+	return resources, nil
+}
+
+func createRoutingSettings(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	log.Printf("Creating Routing Setting")
+	d.SetId("settings")
+	return updateRoutingSettings(ctx, d, meta)
+}
+
+func readRoutingSettings(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getRoutingSettingsProxy(sdkConfig)
+	cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourceRoutingSettings(), constants.DefaultConsistencyChecks, resourceName)
+
+	log.Printf("Reading routing settings: %s", d.Id())
+
+	return util.WithRetriesForRead(ctx, d, func() *retry.RetryError {
+		settings, resp, getErr := proxy.getRoutingSettings(ctx)
+		if getErr != nil {
+			if util.IsStatus404(resp) {
+				return retry.RetryableError(util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("Failed to read Routing Setting %s | error: %s", d.Id(), getErr), resp))
+			}
+			return retry.NonRetryableError(util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("Failed to read Routing Setting %s | error: %s", d.Id(), getErr), resp))
+		}
+
+		resourcedata.SetNillableValue(d, "reset_agent_on_presence_change", settings.ResetAgentScoreOnPresenceChange)
+
+		if diagErr := readRoutingSettingsContactCenter(ctx, d, proxy); diagErr != nil {
+			return retry.NonRetryableError(fmt.Errorf("%v", diagErr))
+		}
+
+		if diagErr := readRoutingSettingsTranscription(ctx, d, proxy); diagErr != nil {
+			return retry.NonRetryableError(fmt.Errorf("%v", diagErr))
+		}
+
+		log.Printf("Read Routing Setting")
+		return cc.CheckState(d)
+	})
+}
+
+func updateRoutingSettings(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	resetAgentOnPresenceChange := d.Get("reset_agent_on_presence_change").(bool)
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getRoutingSettingsProxy(sdkConfig)
+
+	log.Printf("Updating Routing Settings")
```

Comment:
add more context to this log. which would help while troubleshooting.

Reply:
I would include id of the route.  If you can get this fixed, we can merge it

Reply:
This resource controls the routing settings, so there is no ID